From 6f34330862472747d761bae8b405a43368f2a7fc Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Wed, 7 Nov 2018 11:34:22 +0200
Subject: [PATCH 5044/5242] MLK-20043: arm64: dts: imx8mq-evk: add
 dual-display support for B3 board

commit  3e782aca6036ad2d1135e26817ab06af7fcc4be9 from
https://source.codeaurora.org/external/imx/linux-imx.git

For imx8mq-evk board, B4 board change touch/mipi-hdmi connected i2c bus
from i2c1 to i2c3. The default dual-display dts file is for the B4
board.
This patch adds a new dts file to also support dual-display on B3 (or
lower) boards.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    1 +
 .../freescale/fsl-imx8mq-evk-dual-display-b3.dts   |   42 ++++++++++++++++++++
 2 files changed, 43 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dual-display-b3.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 8ad973a..61ff5f1 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -105,6 +105,7 @@ dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-ddr3l-arm2.dtb \
 				 fsl-imx8mq-evk-dcss-rm67191.dtb \
 				 fsl-imx8mq-evk-dcss-rm67191-b3.dtb \
 				 fsl-imx8mq-evk-dual-display.dtb \
+				 fsl-imx8mq-evk-dual-display-b3.dtb \
 				 fsl-imx8mq-evk-ak4497.dtb \
 				 fsl-imx8mq-evk-audio-tdm.dtb \
 				 fsl-imx8mq-evk-drm.dtb \
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dual-display-b3.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dual-display-b3.dts
new file mode 100644
index 0000000..b0c7df7
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dual-display-b3.dts
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8mq-evk-lcdif-adv7535-b3.dts"
+
+/ {
+	display-subsystem {
+		status = "okay";
+	};
+
+	sound-hdmi {
+		status = "okay";
+	};
+};
+
+&irqsteer_dcss {
+	status = "okay";
+};
+
+&dcss {
+	status = "okay";
+};
+
+&hdmi {
+	status = "okay";
+};
+
+&lcdif {
+	status = "okay";
+	max-res = <1280>, <720>;
+};
-- 
1.7.9.5

